### ISSUE TITLE
Unexpected FN&FP

### ISSUE THREAD

**Issue by yangao07** [TimD1/vcfdist#31]
Truth.vcf:
```
chr11   10004574        .       T       TAAAAA  30      .       LCR=0.325011    GT:AD   1|0:1,1
chr11   10004596        .       C       A       30      .       LCR=0   GT:AD   1|0:1,1
```
Query.vcf:
```
chr11   10004596        .       C       A       0       PASS    END=10004596    GT:PS   0|1:9992125
```
Summary.vcf:
```
chr11   10004574        .       T       TAAAAA  .       PASS    .       GT:BD:BC:RD:QD:BK:QQ:SC:SG:PS:PB:BS:FE  1|0:FN:0.166667:6:5:lm:30:38:0:0:8:.:.  .:.:.:.:.:.:.:38:.:.:8:.:.
chr11   10004596        .       C       A       .       PASS    .       GT:BD:BC:RD:QD:BK:QQ:SC:SG:PS:PB:BS:FE  1|0:FN:0.166667:6:5:lm:30:38:0:0:8:.:.  0|1:FP:0.166667:6:5:lm:0:38:0:9992125:8:1:0
```

I found that every time there is an FN, like the chr11:10004574 INS, there will be some unexpected FN&FP, like the chr11:10004596 SNP.

Any suggestions are welcome!
Thank you!

Yan

**Comment by TimD1** [comment]
This may seem unintuitive at first, but this is actually the expected behavior for the current version of vcfdist.

Thanks for starting a discussion on this, because the "right" behavior when evaluating complex variants is still up for debate.

The reason that the query substitution is considered a false positive is that the two truth variants you have shown above are not independent. They are considered part of the same complex variant (because they have the same SC (supercluster) and SG (sync group) tags). As a result, any corresponding query variant(s) must be equivalent to both of these variants (not a subset of them). The evaluation considers the one query variant to be a FP with partial credit of 0.16666, since the edit distance from the truth sequence with the variant has been reduced to 5 (from 6, without the variant). 

The reason that this behavior was selected is to ensure that evaluations are not biased towards specific variant representations. For example, let's assume that all the reference bases between your two variants above are `A`s. Due to the variants, 6 `A`s are added and one `C` is removed. An equally valid variant representation (that I would actually consider more likely), is (10004574 T TAAAAAA), (10004595 AC A). This represents the exact same truth sequence, but now there's not a 1:1 match between any query and truth variants. Would you still want to consider the query substitution a true positive? Many existing benchmarking tools will adjudicate this variant differently, based on the representation of the complex truth variant. vcfdist will not (it will evaluate consistently), since it is almost entirely independent of variant representations. 

The downside of this is that for larger FNs, they may "swallow" nearby SNPs or INDELs that could possibly be considered independent, leading to complex variants where the adjudication is less informative.

In short, yes, there are cases where vcfdist will report FPs and FNs where other tools would not. But unlike those tools, vcfdist will evaluate variants consistently regardless of representation. Once complex variants are nearly or exactly correct, vcfdist will always find them, when other tools with simpler comparison schemes may not. The important thing is that vcfdist has selected an adjudication scheme, performs in-depth comparisons, and is consistent. It may report a lower TP rate than other tools in some cases, but that doesn't make it a worse comparator; for benchmarking you care about the **relative** performance of different tools and pipelines, and vcfdist is always consistent.

**Comment by yangao07** [comment]
Thanks for the detailed explanation! It makes sense to me.
Agree with you on the idea that those variants should be considered together, instead of independently.
vcfdist is definitely better than other tools I have tried.
Great work!

Yan

**Comment by yangao07** [comment]
Hi again, I know this may not be of interest to you:
I am developing a long-read variant calling tool and would like to know more about the variant clustering process in vcfdist.
It would be great if you could share something with me. Or, if you are interested, do you want to chat a little bit?

**Comment by TimD1** [comment]
There's more detail regarding variant clustering in the original paper and on the vcfdist wiki: https://github.com/TimD1/vcfdist/wiki/05-Variant-Clustering

I'd be happy to chat as well, just send an email to  timdunn@umich.edu with an invite to a 15-30 minute Zoom meeting.
